src/Common/DNSResolver.cpp

```diff
@@ -437,6 +437,20 @@ void DNSResolver::addToNewAddresses(const Poco::Net::IPAddress & address)
     impl->new_addresses.insert({address, consecutive_failures});
 }
 
+std::vector<DNSResolver::CacheEntry> DNSResolver::cacheEntries() const
+{
+    std::vector<DNSResolver::CacheEntry> entries;
```

Comment:
Note that methods above lock cache from dropping.
```
void DNSResolver::addToNewAddresses(const Poco::Net::IPAddress & address)
{
    std::lock_guard lock(impl->drop_mutex);
```

Reply:
done